### DISCUSSION TITLE
Exclusion of integer values

### DISCUSSION THREAD

**Discussion by sami-ka** [optuna/optuna#4220] [Q&A] [answered]
Hello, 

Thank you for the super useful package. 

I am currently using it to select a good set of integer values, so I used the following function :

`trial.suggest_int(min_value, max_value)`

The maximum range of `min_value `and `max_value` is approximately from -5 to 5.
However I do not want to sample the 0 value.

The most obvious way for me to do it is to use `trial.suggest_categorical` by giving the integers list without 0, but I guess this comes at the cost of not taking advantage of the natural ordering of integers.

Am I overthinking it? Is there another cleaner way to proceed?
I would be glad to hear your feedback on this!

**Comment by gen740** [comment, marked as answer]
The easiest way to achieve this is to change the `objective` function. For example, if you want to minimize `f(a)` for `-5 ≦ a ≦ 5` except for `a = 0`, let `f(a)` return some big value when `a = 0`.
